### PR TITLE
Handle potential CryptographicException on Windows

### PR DESCRIPTION
**Pull request by 0xced**
Although undocumented, GetRSAPrivateKey may, under unidentified circumstances, throw a "Keyset does not exist" `CryptographicException`.

I had 30+ ASP.NET Core HTTPS development certificate and some of them were somehow corrupted and lead to this exception.

### CODE REVIEW

**Comment by 0xced** [comment]
@javiercn Could you please have a look?

File: src/Shared/CertificateGeneration/WindowsCertificateManager.cs [hunk 1]
```diff
@@ -28,11 +28,20 @@ protected override bool IsExportable(X509Certificate2 c)
...
+                using var key = c.GetRSAPrivateKey();
+                return (key is RSACryptoServiceProvider rsaPrivateKey &&
+                        rsaPrivateKey.CspKeyContainerInfo.Exportable) ||
+                       (key is RSACng cngPrivateKey &&
+                        cngPrivateKey.Key.ExportPolicy == CngExportPolicies.AllowExport);
+            }
+            catch (CryptographicException)
+            {
+                // Although undocumented, GetRSAPrivateKey may, under unidentified circumstances,
+                // throw a "Keyset does not exist" CryptographicException
```
Comment (javiercn) [inline src/Shared/CertificateGeneration/WindowsCertificateManager.cs:42]: We don't think we want to do this.

This I believe could result in the creation of multiple certificates if there is a persistent issue that is causing an exception here.

We'd rather see the tool fail in these cases and get a chance to understand what's going on wrong.

**Comment by javiercn** [comment]
@0xced thanks for your contribution.

Please start with an issue describing the problem to give us some context about what problem you want to solve. Looking at a PR its very hard for us to understand the motivation for changes and the context around them.

If there is a windows configuration that we are not handling and that is causing problems we want to understand the details so that we can try and handle that case specifically more gracefully. Note that we don't intend this tool to behave correctly under all possible configurations since that is untracktable for us. 

We expect it to work on most common user configurations and expect that there are no machine configurations/user permissions that get in the way. We do offer scripts to generate the certificate manually when the tool doesn't work using powershell or openssl.

**Comment by 0xced** [comment]
Unfortunately, I have deleted the certificates that were causing this issue so I can't reproduce it anymore. But the really important point here is that we need to get #25383 merged because as long as it's not merged, we can't **see** any details about the underlying failures. 😕 (And #24559 will also help to get better details by the way.)

I'll close this issue but please follow up on #25383.

**Pull request closed without merging**